src/mbgl/renderer/sources/render_raster_dem_source.hpp

```diff
@@ -6,6 +6,38 @@
 
 namespace mbgl {
 
+enum class DEMTileNeighbors {
+  // 0b00000001
+  Left = 1 << 0,
+  // 0b00000010
+  Right = 1 << 1,
+  // 0b00000100
+  TopLeft = 1 << 2,
+  // 0b00001000
+  TopCenter = 1 << 3,
+  // 0b00010000
+  TopRight = 1 << 4,
+  // 0b00100000
+  BottomLeft = 1 << 5,
+  // 0b01000000
+  BottomCenter = 1 << 6,
+  // 0b10000000
+  BottomRight = 1 << 7
+};
+
+inline unsigned char operator|(unsigned char a, DEMTileNeighbors b) {
```

Comment:
Define `RasterDEMTile::neighboringTiles` of type `DEMTileNeighbors`. Then these operator overrides can uniformly use `DEMTileNeighbors`.

Reply:
I'm not quite sure how to do this because we can't enumerate every single 8-bit number in the enum, so for instance 
```c
neighboringTiles = neighboringTiles | DEMTileNeighbors::Right;
```
can't return a value of type `DEMTileNeighbors` if any other bits on `neighboringTiles` are 1